### PR TITLE
gtk+3 3.24.35

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
On Linux:

```
Configuring gdk.gresource.xml with command
[172](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:173)
  Found pkg-config: /home/linuxbrew/.linuxbrew/opt/pkg-config/bin/pkg-config (0.29.2)
[173](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:174)
  Program glib-compile-resources found: YES (/home/linuxbrew/.linuxbrew/opt/glib/bin/glib-compile-resources)
[174](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:175)
  Program glib-mkenums found: YES (/home/linuxbrew/.linuxbrew/opt/glib/bin/glib-mkenums)
[175](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:176)
  Program glib-mkenums found: YES (/home/linuxbrew/.linuxbrew/opt/glib/bin/glib-mkenums)
[176](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:177)
  Program glib-genmarshal found: YES (/home/linuxbrew/.linuxbrew/opt/glib/bin/glib-genmarshal)
[177](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:178)
  Configuring gdkconfig.h using configuration
[178](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:179)
  Configuring gdkversionmacros.h using configuration
[179](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:180)
  Checking for function "shm_open" : YES 
[180](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:181)
  
[181](https://github.com/Homebrew/homebrew-core/actions/runs/3542500130/jobs/5948150653#step:7:182)
  ../gdk/wayland/meson.build:1:0: ERROR: Non-existent build file 'gdk/wayland/cursor/meson.build'
```

**Comment by chenrui333** [comment]
@iMichka any idea about how to fix on linux?

**Comment by ten0s** [comment]
The problem is **gdk/wayland/cursor/meson.build** that was added in https://gitlab.gnome.org/GNOME/gtk/-/commit/66a199806ceb3daa5e2c7d3a5b45a86007cec46a#93c8a5e8c77e01ecb916156e186ee9ff72105111 is not in the tarball.

I was able to build locally on Ubuntu with the below patch applied. Not sure if this is right thing to do.

```
patch :DATA
```

```
__END__
diff --git a/gdk/wayland/cursor/meson.build b/gdk/wayland/cursor/meson.build
new file mode 100644
index 0000000000000000000000000000000000000000..02d5f2bed8d926ee26bcf4c4081d18fc9d53fd5b
--- /dev/null
+++ b/gdk/wayland/cursor/meson.build
@@ -0,0 +1,12 @@
+wayland_cursor_sources = files([
+  'wayland-cursor.c',
+  'xcursor.c',
+  'os-compatibility.c'
+])
+
+libwayland_cursor = static_library('wayland+cursor',
+  sources: wayland_cursor_sources,
+  include_directories: [ confinc, ],
+  dependencies: [ glib_dep, wlclientdep, ],
+  c_args: common_cflags,
+)
```

**Comment by chenrui333** [comment]
> The problem is **gdk/wayland/cursor/meson.build** that was added in https://gitlab.gnome.org/GNOME/gtk/-/commit/66a199806ceb3daa5e2c7d3a5b45a86007cec46a#93c8a5e8c77e01ecb916156e186ee9ff72105111 is not in the tarball.
> 
> I was able to build locally on Ubuntu with the below patch applied. Not sure if this is right thing to do.
> 
> ```
> patch :DATA
> ```
> 
> ```
> __END__
> diff --git a/gdk/wayland/cursor/meson.build b/gdk/wayland/cursor/meson.build
> new file mode 100644
> index 0000000000000000000000000000000000000000..02d5f2bed8d926ee26bcf4c4081d18fc9d53fd5b
> --- /dev/null
> +++ b/gdk/wayland/cursor/meson.build
> @@ -0,0 +1,12 @@
> +wayland_cursor_sources = files([
> +  'wayland-cursor.c',
> +  'xcursor.c',
> +  'os-compatibility.c'
> +])
> +
> +libwayland_cursor = static_library('wayland+cursor',
> +  sources: wayland_cursor_sources,
> +  include_directories: [ confinc, ],
> +  dependencies: [ glib_dep, wlclientdep, ],
> +  c_args: common_cflags,
> +)
> ```

we can include that build patch, and we also need to submit an issue to notify the upstream as well

@ten0s do you mind filing the upstream issue on that?

**Comment by ten0s** [comment]
@chenrui333 they are aware of it

File: Formula/gtk+3.rb [hunk 1]
```diff
@@ -45,6 +45,8 @@ class Gtkx3 < Formula
     depends_on "libxkbcommon"
     depends_on "wayland-protocols"
     depends_on "xorgproto"
+
+    patch :DATA
```
Comment (SMillerDev) [inline Formula/gtk+3.rb:49]:
```suggestion
    # remove in next release
    patch do
      url "https://gitlab.gnome.org/GNOME/gtk/-/commit/66a199806ceb3daa5e2c7d3a5b45a86007cec46a.patch"
      sha256 "426e0e542b152c50d5cf81ee4710572ee48b98880d1eec4ff12a8c1abe4a079b"
    end
```

Comment (ten0s) [reply Formula/gtk+3.rb:49]: The full patch adds/changes a lot of stuff. Have you tried to build it on Ubuntu?

Comment (chenrui333) [reply Formula/gtk+3.rb:49]: Maybe add a code commit reference for that (it would be removed in the next release though)

Comment (ten0s) [reply Formula/gtk+3.rb:49]: @chenrui333 
The above change doesn't work on Ubuntu. 
I looked around and it seems that the :DATA patch should be put into https://github.com/Homebrew/formula-patches/tree/master/gtk%2B3,  mentioning the original GTK's commit and referenced here with  "Remove in next release".

Comment (chenrui333) [reply Formula/gtk+3.rb:49]: it should be the same thing, putting into the formula-patches or adding directly into the formula.

Comment (chenrui333) [reply Formula/gtk+3.rb:49]: Yeah, I will add that commit reference as code comment after this PR

Comment (chenrui333) [reply Formula/gtk+3.rb:49]: Followup PR #117700

**Pull request closed without merging**